### ISSUE TITLE
Data synchronization

### ISSUE THREAD

**Issue by greuff** [drowe67/codec2#4]
I'm testing the data API right now to check if codec2 fits my intended purpose. I wrote a simple program similar to `freedv_datac1_tx`, that successfully generates a frame that `freedv_datac1_rx` can decode. So far, so good!

Now, to prepare attaching the code to data coming from a sound card, I added some random "noise" in front of the frame with random length. Fed with such data, however, `freedv_datac1_rx` can not always decode the frame. It depends on the number of bytes that I add as "noise". It can deal with a few bytes, or with a number of bytes that is divisible by 200 (other patterns may be possible, but that's what I saw), then the decoder can synchronize to the start of the frame and decode it. Other random numbers of bytes lead to the decoder seemingly not being able to find the frame, and it doesn't get decoded. 

I wonder if this has to do with `freedv_nin`? How would you attach the decoding methods to a real-world sound card, where you don't know where a frame begins? Can FreeDV reliably synchronize to such a random stream?

**Comment by drowe67** [comment]
Yes, FreeDV is used every day with sound cards (e.g. with freedv-gui), and in doing so obtains sync with random timing offsets.  No special treatment is necessary, just feed the received audio in.

Why are you padding the samples with random data?
